src/Runner.Listener/MessageListener.cs

```diff
@@ -74,6 +92,25 @@ public async Task<Boolean> CreateSessionAsync(CancellationToken token)
             string errorMessage = string.Empty;
             bool encounteringError = false;
 
+            var v1Creds = _configStore.GetCredentials();
+            var v2Creds = _configStore.GetV2Credentials();
+            if (v1Creds != null)
+            {
+                _v1CredentialsExists = true;
+            }
+
+            if (v2Creds == null)
+            {
+                if (v1Creds.Scheme == Constants.Configuration.OAuth)
```

Comment:
would v2 ever be null and v1 scheme not oauth?

Reply:
Hosted only has v1 and not oauth.

Reply:
in that case, wouldnt there just be v2 creds?

Reply:
I think the way I name this causes confusion, I should just call them `cred` and `cred.migrate`
A runner that has both `cred` and `cred.migrate` will require to be:
1. self-host runner
2. original configured OAuth with SPS

hosted runner and newly configured runner should always just has `cred`